### PR TITLE
Ability to save/retrieve CCData into/from CCUserDefault

### PR DESCRIPTION
**Pull request by jotel**
I needed to save some binary data into CCUserDefault object... so I created this patch.
It basically moves CCData.\* files from extensions/CCBReader into cocos2dx/cocoa and add some extra methods to CCUserDefault class.

The binary data is stored as base64 encoded string (added encoding function into cocos2dx/support/base64.cpp

I think that other developers might find this useful and you add it to the project... 
I hope that I didn't miss any build script/project... and platforms i'm not familiar will still build  without errors.

Regards
Jaro

### CODE REVIEW

**Comment by dumganhar** [comment]
Did you see travis-ci build fails ? https://travis-ci.org/cocos2d/cocos2d-x/builds/7583580

**Comment by jotel** [comment]
No :( I'll take a look tonight.

Thanks

File: cocos2dx/support/user_default/CCUserDefault.mm [hunk 1]
```diff
@@ -374,6 +375,70 @@ static void deleteNodeByKey(const char *pKey)
...
+#endif
+    
+    NSData *data = [[NSUserDefaults standardUserDefaults] dataForKey:[NSString stringWithUTF8String:pKey]];
+    if (! data)
+    {
+        return defaultValue;
+    }
+    else
+    {
+        unsigned char *bytes = {};
```
Comment (minggo) [inline cocos2dx/support/user_default/CCUserDefault.mm]: This line will cause compiling error on iOS.

Comment (jotel) [inline cocos2dx/support/user_default/CCUserDefault.mm]: It compiles and works for me on iOS ... Can you tell please what the compilation issue you experience ?

Thanks
Jaro

**Comment by jotel** [comment]
Is fixed now... all green;)

Thanks
Jaro

File: cocos2dx/support/user_default/CCUserDefault.mm (hunk 1 above)
Comment (jotel) [inline cocos2dx/support/user_default/CCUserDefault.mm]: You're right... I missed that as I'm using a different C++ Language Dialect
in my project and no compilation errors were reported.

Should be fine now.

Regards
Jaro

Jaroslaw Lewandowski

jotel@jotel.net
http://www.linkedin.com/in/jotel

**Comment by minggo** [comment]
Thank you.

**Comment by dumganhar** [comment]
@minggo 
Does this PR also need to be merged to master branch?
I saw it didn't in master branch yet.

**Comment by minggo** [comment]
Master is used for releasing v2.1 final version. After releasing, it can not be modified directly.
This feature is not needed to be merged into master.

**Pull request merged**